cpp/include/cuvs/core/byte_array.hpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace cuvs::detail {
+
+struct byte_array {
```

Comment:
What do you think about renaming this to something like `byte_ptr` or `byte_arithmetic_ptr`? The reason I'm asking is that this structure semantically looks more like a pointer (dereferencing, pointer arithmetic) rather than a container (no size attribute).